src/main/java/org/opentripplanner/index/model/PatternShort.java

```diff
@@ -1,26 +1,51 @@
 package org.opentripplanner.index.model;
 
-import java.util.Collection;
-import java.util.List;
-
+import com.beust.jcommander.internal.Lists;
 import org.opentripplanner.routing.edgetype.TripPattern;
+import org.opentripplanner.util.model.EncodedPolylineBean;
 
-import com.beust.jcommander.internal.Lists;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
 public class PatternShort {
 
     public String id;
     public String desc;
-    
+    public EncodedPolylineBean geometry;
+
+    /**
+     * Constructor for maintaining backwards compatibility
+     */
     public PatternShort (TripPattern pattern) {
+        this(pattern, null);
+    }
+
+    /**
+     * Constructor for when geometry is passed
+     */
+    public PatternShort (TripPattern pattern, EncodedPolylineBean geometries) {
         id = pattern.code;
         desc = pattern.name;
+        geometry = geometries;
```

Comment:
Doesn't the geometries argument just consist of a mutated version of `Pattern#geometry`? Seems like it would be better to just pass a boolean arg `includeGeometry` and handle the mutation here, e.g.:
```java
geometry = PolylineEncoder.createEncodings(pattern.geometry);
```

Reply:
Wow this is much cleaner thank you very much! Implemented in 94fe2b070ea0ab38e10e88f37b271586688d48bd